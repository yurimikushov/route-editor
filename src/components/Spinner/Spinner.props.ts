type SpinnerProps = {
  className?: string
}

export default SpinnerProps
