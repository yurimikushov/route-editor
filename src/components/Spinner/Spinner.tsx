import { FC } from 'react'
import cn from 'classnames'
import { ReactComponent as SpinnerIcon } from 'icons/spinner.svg'
import SpinnerProps from './Spinner.props'

const Spinner: FC<SpinnerProps> = ({ className }) => {
  return (
    <SpinnerIcon className={cn(className, 'text-indigo-500 animate-spin')} />
  )
}

export default Spinner
