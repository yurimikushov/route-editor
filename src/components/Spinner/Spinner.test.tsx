import renderer from 'react-test-renderer'
import Spinner from './Spinner'

it('should render spinner', () => {
  const tree = renderer.create(<Spinner />).toJSON()
  expect(tree).toMatchSnapshot()
})
