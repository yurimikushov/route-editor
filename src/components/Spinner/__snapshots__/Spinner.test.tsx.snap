// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render spinner 1`] = `
<svg
  className="text-indigo-500 animate-spin"
>
  spinner.svg
</svg>
`;
